File: matlab/Curth_my.m
function Curth_my(Y,K,d,L,R);
Y1=diff(Y);
for k=1:R;
    y=Y(K+(k-1)*d+1:K+(k-1)*d+L);
    sig2=cov(y);  y0=y-mean(y);
    Curth(k)=mean(y0.^4)/(sig2^2);
    V(k)=max(y0)/std(y0);
end;

hF6=figure; 
set(hF6,'Units','Normalized','Pos',[0.2 0.175 0.78 0.55],...
    'Color',[0.8 0.8 0.8]);
%Curth_sm=exp_smooth_twice(Curth',0.5);

subplot(2,1,1);
plot(Curth,'LineWidth',4); grid; hold on; 
title('Фактор куртозиса (динамика эксцесса)','FontSize',14,'FontWeight','Bold');
subplot(2,1,2);
plot(V,'k','LineWidth',4); grid; hold on; 
title('Динамика пик-фактора','FontSize',14,'FontSize',14,'FontWeight','Bold');
ylabel('Пик-фактор','LineWidth',14,'FontSize',14,'FontWeight','Bold');
xlabel('Время (с)','FontSize',14,'FontSize',14,'FontWeight','Bold'); 

File: functions/Curth_my.py
"""
Куртозис и пик-фактор
"""
import numpy as np
from utils import setup_figure, finalize_figure, iter_windows
from config import config


def Curth_my(Y, start_offset, window_shift, window_width, num_windows, fig=None):
    Y = np.array(Y).flatten()
    s = config.plots.style

    Curth = []
    V = []

    for y in iter_windows(Y, start_offset, window_shift, window_width, num_windows):
        sig2 = np.cov(y)
        y0 = y - np.mean(y)

        if sig2 > 0:
            curth = np.mean(y0 ** 4) / (sig2 ** 2)
            std_y0 = np.std(y0)
            v = np.max(y0) / std_y0 if std_y0 > 0 else 0
        else:
            curth = 0
            v = 0

        Curth.append(curth)
        V.append(v)

    fig, [ax1, ax2], standalone = setup_figure(fig, 2, 1, figsize=(12, 8))

    ax1.plot(Curth, linewidth=s.thick_linewidth)
    ax1.grid(True)
    ax1.set_title('Фактор куртозиса (динамика эксцесса)',
                  fontsize=s.title_fontsize, fontweight=s.title_fontweight)

    ax2.plot(V, 'k', linewidth=s.thick_linewidth)
    ax2.grid(True)
    ax2.set_title('Динамика пик-фактора',
                  fontsize=s.title_fontsize, fontweight=s.title_fontweight)
    ax2.set_ylabel('Пик-фактор',
                   fontsize=s.label_fontsize, fontweight=s.label_fontweight)
    ax2.set_xlabel('Время (с)',
                   fontsize=s.label_fontsize, fontweight=s.label_fontweight)

    finalize_figure(fig, standalone)

